landlab/components/network_sediment_transporter/sediment_pulser_at_links.py

```diff
@@ -0,0 +1,320 @@
+    SedimentPulserBase,
+)
+
+_OUT_OF_NETWORK = -2
+
+
+class SedimentPulserAtLinks(SedimentPulserBase):
+    """
+    Send a pulse of parcels to specific links in a channel network
+
+    SedimentPulserAtLinks is instantiated by specifying the network model grid
+    it will pulse the parcels into and the time(s) when a pulse is allowed to accur.
+    It inherits attributes and functions from the SedimentPulserBase class.
+
+    SedimentPulserAtLinks is run (adds parcels to DataRecord) by calling the
+    instance with a list of links and a list of the number of parcels added to each link.
+
+    If parcel attributes are constant with time and uniform
+    accross the basin, these constant-uniform-attirbutes can be defined
+    when SedimentPulserAtLinks is instantiated. If parcel attributes vary
+    with location and time, the user specifies the varying parcel attributes
+    each time the instance is called with a list for each attribute of length
+    equal to the number of links included in the pulse.
+
+
+    authors: Jeff Keck, Allison Pfeiffer, Shelby Ahrendt
+            with help from Eric Hutton and Katy Barnhart
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from landlab import NetworkModelGrid
+    >>> from landlab.components.network_sediment_transporter.sediment_pulser_at_links import SedimentPulserAtLinks
+
+    Create the network model grid the parcels will be added to.
+
+    >>> y_of_node = (0, 100, 200, 200, 300, 400, 400, 125)
+    >>> x_of_node = (0, 0, 100, -50, -100, 50, -150, -100)
+    >>> nodes_at_link = ((1, 0), (2, 1), (1, 7), (3, 1), (3, 4), (4, 5), (4, 6))
+    >>> grid = NetworkModelGrid((y_of_node, x_of_node), nodes_at_link)
+    >>> grid.at_link["channel_width"] = np.full(grid.number_of_links, 1.0)  # m
+    >>> grid.at_link["channel_slope"] = np.full(grid.number_of_links, .01)  # m / m
+    >>> grid.at_link["reach_length"] = np.full(grid.number_of_links, 100.0)  # m
+
+    Define a function that contrains which times a pulse is allowed to occur.
+    This function says a pulse can occur at any time
+
+    >>> def time_to_pulse(time):
+    ...     return True
+
+    Instantiate 'SedimentPulserAtLinks'
+
+    >>> make_pulse = SedimentPulserAtLinks(grid, time_to_pulse=time_to_pulse)
+
+    Run the instance with inputs for the time, link location and number of
+    parcels. Other attributes will use the default values in the base class
+
+    >>> time = 11
+    >>> links = [2, 6]
+    >>> n_parcels_at_link = [2, 3]
+    >>> parcels = make_pulse(time=time, links=links, n_parcels_at_link=n_parcels_at_link)
+
+    Check the element_id of each parcel
+
+    >>> print(parcels.dataset['element_id'].values)
+    [[2]
+     [2]
+     [6]
+     [6]
+     [6]]
+    """
+
+    _name = "SedimentPulserAtLinks"
+
+    _unit_agnostic = False
+
+    _info = {}  # works with the DataRecord
+
+    def __init__(self, grid, time_to_pulse=None, **kwgs):

```

Comment:
Instead of using `**kwgs` specify each of the keywords.

Reply:
@keckje Any chance you have bandwidth to do this in the next few days? I can try, but I'm less familiar with the pulsers.

Reply:
@mcflugen No word from @keckje yet. 

I believe this should be fixed now.